### PR TITLE
Kusto Streaming

### PR DESCRIPTION
**Pull request by ohadbitt**
Kusto Streaming

**Features:**
- the mapping option can now be used to indicate usage of kusto streaming using: "streaming": true

### CODE REVIEW

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java [hunk 1]
```diff
@@ -277,7 +312,12 @@ private static void validateTableAccess(Client engineClient, JSONObject mapping,
                     }
                 }
             }
-        } catch (DataClientException e) {
+            if (hasAccess && shouldCheckStreaming && !isStreamingPolicyEnabled(MAPPING_TABLE, table, engineClient, database)) {
```
Comment (yogilad) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:315]: Do you want to fall back to queued ingestion ?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java [hunk 2]
```diff
@@ -87,22 +90,66 @@ public void handleRollFile(SourceFile fileDescriptor) {
...
             try {
-                client.ingestFromFile(fileSourceInfo, ingestionProps);
+                IngestionResult ingestionResult = client.ingestFromFile(fileSourceInfo, ingestionProps.ingestionProperties);
+                if (ingestionProps.streaming && ingestionResult instanceof IngestionStatusResult) {
+                    // If IngestionStatusResult returned then the ingestion status is from streaming ingest -
+                    // Note: next version it will also contain Queued status - which is OK as getIngestionStatusCollection
+                    // won't cause any data fetch
+                    IngestionStatus ingestionStatus = ingestionResult.getIngestionStatusCollection().get(0);
+                    if (!hasStreamingSucceeded(ingestionStatus) && ingestionStatus.failureStatus.equals(IngestionFailureInfo.FailureStatusValue.Permanent)) {
+                       retryAttempts += ManagedStreamingIngestClient.MAX_RETRY_CALLS - 1;
```
Comment (yogilad) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:100]: This is a little odd.
Isn't this equivalent of breaking out of the loop with some error handling?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java (hunk 1 above)
Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:315]: i don't

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java (hunk 2 above)
Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:100]: i think only misstake here is the check for permanent - ill remove it

Comment (yogilad) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:100]: ok, but why are you moving the retry-counter all the way to the end?
What's the reason behind it

Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:100]: oh ill add a comment about it (this max is coming from the client - the managed streaming does some retries by itself )
actually i dont think we would reach this code as we probably wont fail on queued fallback - maybe we should instead make the retries for streaming configureable

**Pull request merged**